eis_toolkit/vector_processing/rasterize_vector.py

```diff
@@ -0,0 +1,111 @@
+from typing import Optional, Tuple
+
+import geopandas as gpd
+import numpy as np
+from rasterio import features, profiles, transform
+
+from eis_toolkit import exceptions
+
+
+def rasterize_vector(
+    geodataframe: gpd.GeoDataFrame,
+    resolution: float,
+    value_column: Optional[str] = None,
+    default_value: float = 1.0,
+    fill_value: float = 0.0,
+    base_raster_profile: Optional[profiles.Profile] = None,
```

Comment:
The main entrypoint takes in a `rasterio.profiles.Profile` rather than a `rasterio.io.DatasetReader` which is more often the case in `eis_toolkit`. I mentioned this in #120. I believe it is appropriate in just this function as the array data from the raster is not needed but there might be some value in always passing the reader instead to avoid people having to figure out what is the `profiles.Profile`. Small details of course...